tests/tests/test-class-db-driver-wpdb.php

```diff
@@ -98,6 +111,190 @@ public function test_table_names() {
 		$this->assertEquals( array( $this->driver->table, $this->driver->table_meta ), $table_names );
 	}
 
+	// Test that the purge_storage() function removes database records as expected.
+	public function test_purge_storage() {
+		global $wpdb;
+
+		if ( is_multisite() ) {
+			$this->markTestSkipped( 'This test requires single site.' );
+		}
+
+		// Test cases override table dropping queries to prevent them
+		// from being executed, but we need actual dropping here.
+		foreach ( $GLOBALS['wp_filter']['query'] as $priority => $filters ) {
+			foreach ( $filters as $filter => $data ) {
+				if ( false !== strpos( $filter, '_drop_temporary_table' ) ) {
+					unset( $filters[$filter] );
+					$GLOBALS['wp_filter']['query'][$priority] = $filters;
+				}
+			}
+		}
```

Comment:
This is needed because our `dbdelta` SQL is using `CREATE TABLE`, but the drops are turned into `DROP TEMPORARY TABLE` by default by the WP test case, breaking the drops as the temporary tables were never created (they were regular tables instead).